api/bibliotheca.py

```diff
@@ -1346,25 +1350,59 @@ def create_default_start_time(self, _db, cli_date):
     def catch_up_from(self, start, cutoff, progress):
         added_books = 0
         i = 0
-        one_day = timedelta(days=1)
+        init_minute_span = 5
+        timespan_to_check = timedelta(minutes=init_minute_span)
+
+        # Not having events in a time period is not considered to be
+        # an error.
+        no_events_error = False
+        # But in this case, we do want to consider not getting events as
+        # an error. Each subsequent time this is run after the first time,
+        # the timespan to check for event increases by 5 minutes until we
+        # reach a 70-hour timespan to check for events.
+        if (progress.counter > 0):
+            no_events_error = True
+            minutes = init_minute_span * progress.counter
+            timespan_to_check = timedelta(minutes=minutes)
```

Comment:
The timespan to check is the entire time between `start` and `cutoff` -- unless that time is more than 70 hours. If it's more than 70 ours, timespan_to_check should be bumped back down to 5 minutes, no_events_error should be set to False, and progress.counter should be set to zero.

Basically, timespan_to_check grows until it reaches 70 hours, at which point we go back into "initial import" mode.

I was imprecise when I said that the time slice would get five minutes bigger every time--that's because the script is run every five minutes. It shouldn't be necessary to explicitly do that.

Does this make sense?

Reply:
Yes, it does.